api/jobs.go

```diff
@@ -513,10 +512,6 @@ func (u *UpdateStrategy) Canonicalize() {
 		u.MinHealthyTime = d.MinHealthyTime
 	}
 
-	if u.AutoRevert == nil {
-		u.AutoRevert = d.AutoRevert
-	}
```

Comment:
Not sure I follow the reason for this removal.  I might be missing some context.  Also, if this is a bug, can we have a test case please?

Reply:
there's a comment out of view at the top of the struct that explains the reason: with booleans in the default struct, a missing value is masked by a false, which seems like an intentional false when merging update stanzas together

Reply:
I see, thanks.  It would be useful to point to the comment that in the PR description to aid the reviewer, who isn't as familiar with the code.  Also, I don't quite understand it: If user leaves field unset in config but the default value of the field is `true`, I would have expected `Canonicalize` to render the field as `true`?   Where else do apply default values?

Also, the test is failing.